docs/website/generate_extra_files.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Generates documentation files that require extra steps outside of mkdocs.
+#
+# Typical usage:
+#   docs/website$ ./generate_extra_files.sh
+#   docs/website$ ./mkdocs serve
+#
+# This script should be re-run whenever the generated files change, e.g. when
+# updating MLIR dialect .td files.
+
+set -xeuo pipefail
+
+THIS_DIR="$(cd $(dirname $0) && pwd)"
+REPO_ROOT="$(cd $THIS_DIR/../../ && pwd)"
+DOCS_BUILD_DIR="${DOCS_BUILD_DIR:-${REPO_ROOT}/build-docs}"
+
+mkdir -p "${DOCS_BUILD_DIR}"
+
+# Build `iree-doc` CMake target. This requires the LLVM submodule and can take
+# several minutes, as it builds `iree-tblgen`.
```

Comment:
How about fetching iree-tblgen from the previously-completed postsubmit build?

Reply:
Considered that... seems complicated, and I was trying to keep this fairly simple (to the point where someone could fork it without needing to untangle a bunch of supporting CI scripts... not that anyone _would_, just as a general litmus test).

Tablegen _is_ faster to build than the rest of LLVM (no codegen, targets, etc.), but it's still pretty slow by GitHub standard runner CI standards, thanks to `LLVMSupport`

Reply:
40-60s before: https://github.com/openxla/iree/actions/workflows/publish_website.yml
~5 minutes (sample size 1) after: https://github.com/ScottTodd/iree/actions/runs/5271864329/jobs/9533338273

I'm pretty comfortable with that, considering the filtering and standard linux runner (not a large hosted runner), but I could explore options like that to get the time down if you think it's worth it.

Reply:
I could also enable ccache...

Reply:
Ah not having to build all of LLVM makes a big difference indeed. 5 minutes on the cheapo runners isn't too bad. To enable ccache you should run in the docker container

Reply:
Another, related idea: local builds could pass `IREE_BIN_DIR` to avoid needing to create a separate build directory and rebuild, if a developer has already built `iree-tblgen`.